src/commands/server/delete.ts

```diff
@@ -74,30 +76,33 @@ export default class Delete extends Command {
       tasks.add(cheTasks.deleteNamespace(flags))
     }
 
-    const cluster = KubeHelper.KUBE_CONFIG.getCurrentCluster()
-    if (!cluster) {
-      throw new Error('Failed to get current Kubernetes cluster. Check if the current context is set via kubect/oc')
-    }
-
-    if (!flags['skip-deletion-check']) {
-      const confirmed = await cli.confirm(`You're going to remove Eclipse Che server in namespace '${flags.chenamespace}' on server '${cluster ? cluster.server : ''}'. If you want to continue - press Y`)
-      if (!confirmed) {
-        this.exit(0)
+    if (await this.isDeleteConfirmed(flags)) {
+      try {
+        await tasks.run()
+        cli.log(getCommandSuccessMessage(this, ctx))
+      } catch (error) {
+        cli.error(error)
       }
-    }
-
-    try {
-      await tasks.run()
-      cli.log(getCommandSuccessMessage(this, ctx))
-    } catch (error) {
-      cli.error(error)
+    } else {
+      this.exit(0)
     }
 
     notifier.notify({
       title: 'chectl',
       message: getCommandSuccessMessage(this, ctx)
     })
+  }
+
+  async isDeleteConfirmed(flags: any): Promise<boolean> {
+    const cluster = KubeHelper.KUBE_CONFIG.getCurrentCluster()
+    if (!cluster) {
+      throw new Error('Failed to get current Kubernetes cluster. Check if the current context is set via kubectl/oc')
+    }
+
+    if (!flags.yes && !flags['skip-deletion-check']) {
```

Comment:
Can we:
1. Deprecate `skip-deletion-check` flag and print warning message if it was used
2. Handle the logic at the command beginning to have only one parameter to analyse then